File: core/src/main/java/glitch/api/ws/events/IEvent.java
package glitch.api.ws.events;

import glitch.service.AbstractWebSocketService;

import java.time.Instant;
import java.util.UUID;

public interface IEvent<S extends AbstractWebSocketService<S>> {
    S getClient();
    Instant getCreatedAt();
    UUID getEventId();
}

File: core/src/main/kotlin/glitch/api/ws/events/IEvent.kt
package glitch.api.ws.events

import glitch.service.ISocketService
import java.time.Instant
import java.util.*

/**
 * Event Interface
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
interface IEvent<S : ISocketService<S>> {
    /**
     * the [glitch.GlitchClient]
     */
    val client: S
    /**
     * Event creation time
     */
    val createdAt: Instant
        get() = Instant.now()

    /**
     * unique Event ID
     */
    val eventId: UUID
        get() = UUID.randomUUID()
}
